lib/aws-xray-sdk/model/entity.rb

```diff
@@ -155,6 +155,8 @@ def to_h
       if (a = annotations.to_h) && !a.empty?
         h[:annotations] = a
       end
+      # make sure @metadata is initialized before evaluating it, to prevent warning `variable @metadata not initialized`
+      metadata
```

Comment:
I should have paid more attention to what `metadata` is doing

```
    def metadata(namespace: :default)
      @metadata ||= Metadata.new(self)
      @metadata.sub_meta(namespace)
    end
```
While `@metadata` is containing a `Metadata` object, the metadata method, is actually returning a `SubMeta` object. Explaining why I broke things, replacing one with the other.

So the new idea here, is just to initialize `@metadata` when needed, before the condition, doing so preventing the warning.

Tested it locally, let's see if this time it works